### PR TITLE
steps: Add GitTag buildstep

### PR DESCRIPTION
**Pull request by chapipo**
Add a buildstep to be able to tag a local Git repository.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/steps/source/git.py [hunk 1]
```diff
@@ -697,3 +697,78 @@ def _doPush(self):
...
+
+        if self.messages and not isinstance(self.messages, list):
+            bbconfig.error('GitTag: messages should be a list')
+
+    @defer.inlineCallbacks
+    def run(self):
+        self.stdio_log = yield self.addLog("stdio")
+        gitInstalled = yield self.checkTagSupport()
+
+        if not gitInstalled:
```
Comment (p12tic) [inline master/buildbot/steps/source/git.py]: Coverage says this branch is not covered. Could we have a small unit test for this condition?

**Comment by chapipo** [comment]
I don't know if the adding the function `checkTagSupport` in `GitStepMixin` is the good way. Maybe we could rename the `checkBranchSupport` to something else ?

File: master/buildbot/steps/source/git.py (hunk 1 above)
Comment (chapipo) [reply master/buildbot/steps/source/git.py]: Yeah I spotted it, I will do it and add some more case to current unit test to cover identical branch in other buildstep.

**Comment by p12tic** [comment]
Indeed, I missed that. Since #4558 we can freely rename internal APIs. I think `checkFeatureSupport` could be a candidate for a better name.

**Comment by chapipo** [comment]
Function `checkBranchSupport`renamed to `checkFeatureSupport` and unit tests added.

**Pull request merged**